compiler/rustc_errors/src/translation.rs

```diff
@@ -64,8 +68,70 @@ pub trait Translate {
             DiagnosticMessage::Str(msg) | DiagnosticMessage::Eager(msg) => {
                 return Ok(Cow::Borrowed(msg));
             }
+            DiagnosticMessage::FluentRaw(msg) => {
+                // FIXME(yukang): calculate the `slug` from the raw fluent content,
+                // The fluent resources are generated by a simple standalone visitor:
+                // https://github.com/chenyukang/fluent-utils/blob/main/src/visitor.rs#L13-L97
```

Comment:
I think I still prefer some solution where the compiler is able to emit the reference `ftl` rather than needing another tool to extract it. I know that we tried to do this by writing static variables into a section, but we might also be able to do it from the proc macro by writing to the filesystem (which isn't ideal) based on an environment variable.